components/sup/src/cli.rs

```diff
@@ -1,25 +1,29 @@
 use clap::App;
 use hab::cli::sup_commands;
+use habitat_common::FeatureFlag;
 
-pub fn cli<'a, 'b>() -> App<'a, 'b> { sup_commands() }
+pub fn cli<'a, 'b>(feature_flags: FeatureFlag) -> App<'a, 'b> { sup_commands(feature_flags) }
 
 #[cfg(test)]
 mod test {
     use super::cli;
     use clap::ErrorKind;
+    use habitat_common::FeatureFlag;
+
+    fn no_feature_flags() -> FeatureFlag { FeatureFlag::empty() }
 
     macro_rules! assert_cli_cmd {
-        ($test:ident, $cmd:expr, $( $key:expr => $value:tt ),+) => {
+        ($test:ident, $flags:expr, $cmd:expr, $( $key:expr => $value:tt ),+) => {
             #[test]
             fn $test() {
-                assert_cmd!(cli(), $cmd, $( $key => $value ),+ );
+                assert_cmd!(cli($flags), $cmd, $( $key => $value ),+ );
```

Comment:
I figured it would be easier for Future Us to make the macro know how to handle the flags now. That way, it's very obvious what you need to do to write a new test, and doesn't require you to dig into the macro to figure out what you need to change in order to add a new test that _does_ require feature flags.

(Alternatively, I'll look into whether adding a few tests here would be useful.)

Reply:
Actually, we've got tests elsewhere, so I won't add others here.

I think I'll leave the other changes as they are. Having the `no_feature_flags()` bit explicitly in the tests is informative by itself. It also maintains a kind of parity between the tests that use `assert_cli_cmd!` and the ones that don't.

In general, though, I think tests for things that are behind feature flags can still be useful, since they can help ensure that you get the right behavior, and can continue to be used if and when the flagged feature becomes a "real" one.